File: OOPS/rectangleencapsulation-project/src/com/techlabs/rectangle/Rectangle.java
package com.techlabs.rectangle;

public class Rectangle {

	private int width;
	private int height;
	private String bgColor;
	private static final int MAX=100;
	private static final int MIN=0;
	private static final String RED="red";
	private static final String GREEN="green";
	private static final String BLUE="blue";
	
	public void setWidth(int width) {
		this.width=validateValues(width);
	}
	
	public void setHeight(int height) {
		this.height=validateValues(height);
	}
	

	public void setBgColor(String bgColor) {
		this.bgColor = validateBgColor(bgColor.toLowerCase());
	}
	public int getWidth() {
		return width;
	}


	public int getHeight() {
		return height;
	}

	
	public String getBgColor() {
		return bgColor;
	}

	
	
	private int validateValues(int value)
	{
		if(value>MAX)
		{
			return MAX;
		}
		if(value<MIN)
		{
			return MIN;
		}
		return value;
		
	}
	
	
	private String validateBgColor(String bgColor)
	{
		switch(bgColor)
		{
		case RED:return RED; 
		case BLUE:return BLUE;
		case GREEN:return GREEN;	
		}
		return null;
	}
	public int calculateArea()
	{
		int area;
		area=width*height;
		return area;
	}
	
}

File: CSharpOOPS/rectangleencapsulation/rectangleencapsulation/rectangle/Rectangle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace rectangleencapsulation.rectangle
{
    class Rectangle
    {
        private int _width;
        private int _height;
        private _bgcolor _bgColor;

        public enum _bgcolor
        {
            RED,GREEN,BLUE
        }
        private const int MAX = 100;
        private const int MIN = 0;

        public int Width
        {
            get
            {
                return _width;
            }
            set
            {
                _width = value;
            }
        }
        public int Height
        {
            get
            {
                return _height;
            }
            set
            {
                _height = value;
            }
        }

        public _bgcolor BgColor
        {
            get
            {
                return _bgColor;
            }
            set
            {
                _bgColor = value;
            }
        }

        private int validateValues(int values)
        {
            if (values > MAX)
            {
                return MAX;
            }
            if (values < MIN)
            {
                return MIN;
            }
            return values;

        }

        public int calculateArea()
        {
            int area;
            area = _width * _height;
            return area;
        }

    }
}
